lib/openstax/rescue_from.rb

```diff
@@ -0,0 +1,136 @@
+require 'openstax/rescue_from/configuration'
+
+module OpenStax
+  module RescueFrom
+    class << self
+      def perform_rescue(exception:, listener: MuteListener.new)
+        proxy = ExceptionProxy.new(exception)
+
+        listener.before_openstax_exception_rescue(proxy)
+        register_exception(exception.class)
+        log_system_error(proxy)
+        send_notifying_exceptions(proxy, listener)
+        finish_exception_rescue(proxy, listener)
+      end
+
+      def register_exception(exception, options = {})
+        unless registered_exceptions.include?(exception.name)
```

Comment:
K, gotta think about how to do this one... the rest of the comments have been fulfilled